faunadb-java-dsl/src/main/java/com/faunadb/client/types/Field.java

```diff
@@ -71,6 +73,53 @@ public CollectionCodec(Path path, Field<A> field) {
       };
   }
 
+  private static final class MapCodec<A> implements Codec<Map<String, A>> {
+    private final Path path;
+    private final Field<A> field;
+
+    public MapCodec(Path path, Field<A> field) {
+      this.path = path;
+      this.field = field;
+    }
+
+    @Override
+    public Result<Map<String, A>> decode(Value input) {
+      return input.to(OBJECT).<Map<String, A>>flatMap(toMap);
+    }
+
+    @Override
+    public Result<Value> encode(Map<String, A> value) {
+      throw new UnsupportedOperationException("not implemented");
+    }
+
+    private final Function<Map<String, Value>, Result<Map<String, A>>> toMap =
+      new Function<Map<String, Value>, Result<Map<String, A>>>() {
+
+        @Override
+        public Result<Map<String, A>> apply(Map<String, Value> values) {
+          Map<String, A> success = new LinkedHashMap<>();
+          List<String> failures = new ArrayList<>();
+
+          for (String key : values.keySet()) {
```

Comment:
Is it more efficient to get an entry set instead of getting the key and then looking it up?

Reply:
Could be. Done.